core/src/IdToken.cs

```diff
@@ -43,6 +43,7 @@ internal class IdTokenClaim
         public const string PreferredUsername = "preferred_username";
         public const string Name = "name";
         public const string HomeObjectId = "home_oid";
+        public const string Upn = "upn";
```

Comment:
This is v2 IdToken - it does not contain "upn" claim, 
that claim is from v1 idToken

Reply:
Would unique_name be better?

Reply:
Msal is SDK for V2 endpoints, so it is supposed to receive itTokens in v2 format (which is more aligned with OpenID Connec spec) https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens